packages/runtime-common/schema-analysis-plugin.ts

```diff
@@ -407,3 +487,25 @@ export function isInternalReference(
 ): classReference is InternalReference {
   return classReference && classReference.type === 'internal';
 }
+
+function insertOrReplace(item: Declaration, arr: Declaration[]) {
+  let localName = item.localName;
+  let existingDeclaration = arr.find((i) => {
+    return i.localName === localName;
+  });
+  if (existingDeclaration) {
+    console.log(
+      `declaration of name ${localName} already exists. It is of type ${existingDeclaration.type}. Attempting to replace with type ${item.type}`,
+    );
+    if (
+      item.type === 'possibleCardOrField' &&
+      existingDeclaration.type === 'reexport'
+    ) {
+      let index = arr.indexOf(existingDeclaration);
+      arr.splice(index, 1, item);
+    }
+  } else {
+    arr.push(item);
+  }
+  return arr;
+}
```

Comment:
intentionally made this function so coreVisitor and reExport visitor don't make redundant declarations. If a code declaration exist, we will prefer that one